front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/ItineraryMarkersV2.tsx

```diff
@@ -0,0 +1,174 @@
+import React, { useCallback, useMemo } from 'react';
+
+import type { Position } from '@turf/helpers';
+import cx from 'classnames';
+import type { Map } from 'maplibre-gl';
+import { Marker } from 'react-map-gl/maplibre';
+import { useSelector } from 'react-redux';
+
+import destinationSVG from 'assets/pictures/destination.svg';
+import originSVG from 'assets/pictures/origin.svg';
+import viaSVG from 'assets/pictures/via.svg';
+import { useOsrdConfSelectors } from 'common/osrdContext';
+import type { PathStep } from 'reducers/osrdconf/types';
+import { getNearestTrack } from 'utils/mapHelper';
+
+enum MARKER_TYPE {
+  ORIGIN = 'origin',
+  VIA = 'via',
+  DESTINATION = 'destination',
+}
+
+type MarkerInformation = {
+  marker: PathStep;
+  coordinates: number[] | Position;
+  imageSource: string;
+} & (
+  | {
+      type: MARKER_TYPE.ORIGIN | MARKER_TYPE.DESTINATION;
+    }
+  | {
+      type: MARKER_TYPE.VIA;
+      index: number;
+    }
+);
+
+const formatPointWithNoName = (
+  lineCode: number,
+  lineName: string,
+  trackName: string,
+  markerType: MarkerInformation['type']
+) => (
+  <>
+    <div className="main-line">
+      <div className="track-name">{trackName}</div>
+      <div className="line-code">{lineCode}</div>
+    </div>
+    <div className={cx('second-line', { via: markerType === MARKER_TYPE.VIA })}>{lineName}</div>
+  </>
+);
+
+const extractMarkerInformation = (
+  origin: PathStep | null,
+  vias: (PathStep | null)[],
+  destination: PathStep | null
```

Comment:
Use directly pathSteps and a reduce
```suggestion
  pathSteps:  (PathStep | null)[]
```

Reply:
Done